service/client/client.js

```diff
@@ -717,6 +717,14 @@ export default class Client {
             data = await response.json();
         } else {
             data = await response.text();
+            if (isResponseInvalid(data)) {
```

Comment:
Why not just ignore the content type then and return that the response is invalid if `data = await response.json();` fails? If we're treating everything that isn't json as invalid, then checking everything else is unnecessary